bootcompiler/src/main/scala/org/mozartoz/bootcompiler/ast/CommonStatExpr.scala

```diff
@@ -43,9 +43,12 @@ trait ProcFunExpression extends StatOrExpr {
 }
 
 trait CallCommon extends StatOrExpr {
+  /** 1 for tail call, 2 for tail call to current proc */
+  var tail_self = 0
```

Comment:
Mmh, so to ease interaction with Java?
How about some methods or public fields so that we can do CallKind#isTailCall() and CallKind#isSelfCall() ?